src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_ecc_import_export.c

```diff
@@ -7,6 +7,26 @@
 #include "pal_jni.h"
 #include "pal_utilities.h"
 
+static jobject CryptoNative_CreateBigInteger(uint8_t* data, int32_t len)
+{
+    JNIEnv* env = GetJNIEnv();
+    jbyteArray buffArray = (*env)->NewByteArray(env, len);
+    (*env)->SetByteArrayRegion(env, buffArray, 0, len, (jbyte*)data);
+    jobject bigNum;
+    if (data[0] & 0x80)
+    {
+        // If the top bit is set, we need to create a BigInteger with the sign-and-magnitude constructor.
+        bigNum = (*env)->NewObject(env, g_bigNumClass, g_bigNumCtorWithSign, 1, buffArray);
```

Comment:
My initial reaction was "why is this working for OpenSSL, then?", but:

> BN_bin2bn() converts the positive integer in big-endian form of length len at s into a BIGNUM and places it in ret. If ret is NULL, a new BIGNUM is created.

So the most correct action would be to fix the existing routines to always do a positive import.

Similarly, the data export is

> BN_bn2bin() converts the absolute value of a into big-endian form and stores it at to. to must point to BN_num_bytes(a) bytes of memory.


Being consistent with those seems easier while you're in the hybrid state.

Reply:
Actually, looks like you did change that one to always be unsigned.  So how is this version different (besides being longer, since I think both the if and else could both call the if)?

Reply:
This was my first attempt to work around the leading-zero-byte issue. I'll go back to reusing the bignum PAL apis.